File: frontend/src/api/country.ts
import http from './http'

export interface Country {
  id: number
  owner_id?: number | null
  /** 中文名，如 日本 */
  name_zh: string
  /** 英文名，如 Japan */
  name_en?: string | null
  /** 国家代码，如 JP */
  code?: string | null
  remark?: string | null
  sort_order: number
  /** 后端拼好的「中文 / English」 */
  label: string
  created_at: string
  updated_at: string
}

export interface CountryPayload {
  name_zh: string
  name_en?: string | null
  code?: string | null
  remark?: string | null
  sort_order?: number
}

export const countryApi = {
  list: () => http.get<unknown, Country[]>('/countries'),
  create: (body: CountryPayload) => http.post<unknown, Country>('/countries', body),
  update: (id: number, body: Partial<CountryPayload>) =>
    http.put<unknown, Country>(`/countries/${id}`, body),
  remove: (id: number) => http.delete<unknown, { ok: boolean }>(`/countries/${id}`),
}

File: backend/app/models/country.py
from __future__ import annotations

from sqlalchemy import ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base, TimestampMixin


class Country(Base, TimestampMixin):
    """国家字典：达人的国家分类，中英文各存一份，代码用于对齐外部数据。"""

    __tablename__ = "countries"

    owner_id: Mapped[int | None] = mapped_column(
        ForeignKey("users.id", ondelete="SET NULL"), nullable=True, index=True
    )
    #: 中文名，如「日本」
    name_zh: Mapped[str] = mapped_column(String(128), nullable=False, index=True)
    #: 英文名，如「Japan」
    name_en: Mapped[str | None] = mapped_column(String(128), nullable=True, index=True)
    #: 国家代码（ISO 3166-1 alpha-2 习惯写法，如 JP），抓取结果按它对齐
    code: Mapped[str | None] = mapped_column(String(16), nullable=True, index=True)
    remark: Mapped[str | None] = mapped_column(Text, nullable=True)
    sort_order: Mapped[int] = mapped_column(Integer, default=0, nullable=False)

    @property
    def label(self) -> str:
        """列表里展示用的「中文 / English」。"""
        return f"{self.name_zh} / {self.name_en}" if self.name_en else self.name_zh
